### PR TITLE
✅  Box Selection Plugin Test

### PR DESCRIPTION
**Pull request by aldebaran30701**
Add test cases for select in box plugin as per #1023 

Tests: 
- select nothing
- select one node
- select two nodes and one transaction

All test were run with two boolean flags toggled in every state, and camera mix of 1.0 and 0.0.

### CODE REVIEW

File: CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    /**
+     * Run a box select plugin test
+     *
+     * @param isAdd
+     * @param isToggle
+     * @throws InterruptedException
+     * @throws PluginException
+     * @throws IOException
+     */
+    public void boxSelect(final boolean isAdd, final boolean isToggle) throws InterruptedException, PluginException, IOException {
```
Comment (arcturus2) [inline CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java:134]: In the `testEdit` method it looks like all 4 permutations of these 2 parameters are tested but I noticed that `isAdd` and `isToggle` is not used by the `boxSelect` method. Can you clarify?

**Review by arcturus2 — COMMENTED** [review]
I was also wondering with `boxSelect` whether the method could be broken up into smaller tests like `testBoxSelectWithNothingSelected`, `testBoxSelectWith...` for the different scenarios? If the purpose of `boxSelect` is something else feel free to ignore this comment.

File: CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java (hunk 1 above)
Comment (aldebaran30701) [reply CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java:134]: Great spotting, yeah I refactored it but forgot to add the parameters into it! Will make some changes to it - you are correct on your assumption of the `boxSelect()` method 👍

Comment (arcturus2) [reply CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPluginNGTest.java]: Awesome love your work @aldebaran30701

**Review by arcturus2 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**